### DISCUSSION TITLE
How can I detect the power monitoring capability ?

### DISCUSSION THREAD

**Discussion by wjohn007** [arendst/Tasmota#22335] [Q&A] [answered]
I intend to write a Berry script for multiple controllers. 
Not all of them support power monitoring.
How can I determine whether the specific controller supports power monitoring? 
I would prefer to use the Berry Module Energy for this.

Thank you very much for the support.
John

**Comment by sfromis** [comment]
That depends on what you even mean by "controller". If what you mean is a Tasmota device correctly configured for energy monitoring, the data will be available with other sensor data.

**Comment by wjohn007** [comment]
The Berry script is intended to detect whether energy monitoring is configured and available for the current Tasmota-Controller.

**Reply by sfromis** [reply, marked as answer]
Well, as I mentioned, energy monitoring numbers will be available as part of the sensor data. In Berry, you could use `tasmota.read_sensors` and check if the JSON (parsed into a `map`) contains the `ENERGY` object.

**Comment by wjohn007** [comment]
Thank you for the hint.
This solves my requirement:

`json.load(tasmota.read_sensors()).contains('ENERGY')`
